munch-match/src/services/MapsService.js

```diff
@@ -1,118 +1,129 @@
-
 /**
- * This script interacts with the google maps places api to retrieve the necessary information. 
- * 
+ * This script interacts with the google maps places api to retrieve the necessary information.
+ *
  * getResults() - Takes input the query type and the users coordinate information to retrieve information such as place name, rating etc..
- * fetchAllDetails() - 
+ * fetchAllDetails() -
  */
 
 
 //Creates the google map variable
 const google = window.google;
 
-function getResults(queryType, lat, lng) {
-  //Stores the user's location as google.maps.LatLng (Latitude and Longitude)
-  //Change location variables to props variables
-  const userLocation = new google.maps.LatLng(lat, lng);
+function getResults(queryType, lat, lng, ratingRange, priceRanges) {
+    //Stores the user's location as google.maps.LatLng (Latitude and Longitude)
+    //Change location variables to props variables
+    const userLocation = new google.maps.LatLng(lat, lng);
+
+    //The parameters for requesting nearby locations for places to eat
+    let request = {
+        //Change query to button input
+        query: queryType,
+        location: userLocation,
+        //Distance from userLocation which is in meters
+        radius: 5000,
+        openNow: true,
+        type: "restaurant",
+    };
+    //Creates the variable on where it will store the output of the requested data should be
+    let service = new google.maps.places.PlacesService(
+        document.createElement("div")
+    );
+
+    //Gets the data and retrieves the data to the callback function
+    return new Promise((resolve, reject) => {
+        service.textSearch(request, (results, status) => {
+            if (status === google.maps.places.PlacesServiceStatus.OK) {
+                let extractedResults = [];
+                (async () => {
+                    try {
+                        let detailed_results = await fetchAllDetails(results, [], service);
+                        // Code to run after fetchAllDetails has completed
+                        console.log("All details fetched and processed.");
+                        const today = new Date();
+                        const days = [6, 0, 1, 2, 3, 4, 5];
+                        const dayOfWeek = today.getDay();
+
+                        const ratingFilteredResults = detailed_results.filter(result => {
```

Comment:
nothing crazy but maybe just some code comments
// Filter out results based on the provided rating range